### ISSUE TITLE
--sudo-user generates permission errors

### ISSUE THREAD

**Issue by waldt** [ansible/ansible#353]
--sudo-user doesn't work

Testing against local server:

user1@master:~$ ansible 127.0.0.1 -m ping -u user1 -s --sudo-user=user2
127.0.0.1 | FAILED => failed to parse: /bin/bash: /home/user1/.ansible/tmp/ansible.gZgKLr/ping: Permission denied
//bin/bash: /home/user1/.ansible/tmp/ansible.gZgKLr/ping: Permission denied
/ping
127.0.0.1 | FAILED => failed to parse: /bin/bash: /home/user1/.ansible/tmp/ansible.gZgKLr/ping: Permission denied
//bin/bash: /home/user1/.ansible/tmp/ansible.gZgKLr/ping: Permission denied
/ping

This works, however:

user1@master:~$ ansible 127.0.0.1 -m ping -u user1 -s --sudo-user=user1
127.0.0.1 | success >> {
    "ping": "pong"
}

**Comment by mpdehaan** [comment]
are you using the latest version of the devel branch?

**Comment by waldt** [comment]
Yes, the latest version of the devel branch.

**Comment by mpdehaan** [comment]
hmm, so it's using the tempdir for user1 not the sudo user, while trying to execute as the other user.

umask possibly?

**Comment by mpdehaan** [comment]
I'm pondering whether the chmod just needs to be a+x instead of just +x in this case.  Several of us are ok, so this seems to be it.

**Comment by sfromm** [comment]
Should sudo use -i (simulate initial login) so as to limit the potential conflict between different user settings?

**Comment by mpdehaan** [comment]
Maybe experiment and see if it makes it better? 

Trying one/both seem reasonable.

**Comment by sfromm** [comment]
I can try to replicate.  waldt, can you describe your environment so that I can try to replicate the problem?

**Comment by sfromm** [comment]
I can replicate it.  The problem isn't umask per se.  The two users i tried with both have umask of 002.  The problem is that ~user1/.ansible/tmp/ansible.NNNNN is created 700.  But, I can see this being a problem if anything in that path is restricted such that user2 can't read it.  For example,  I'd expect a similar failure if ~user1/ is 710 and user2 is not root.

**Comment by sfromm** [comment]
Bah, I misread what to chmod above.  Let me try again.

**Comment by sfromm** [comment]
Confirmed.  mktemp, per manpage, creates directories as _u+rwx, minus umask restrictions_.  I'll file a pull request in a bit.

**Comment by sfromm** [comment]
waldt, please try the fix in commit 5e5b4a9 and let me know if that resolves the issue for you.

**Comment by mpdehaan** [comment]
closing, let me know if unfixed

**Comment by waldt** [comment]
I can confirm that the issue is fixed now. Many thanks!
